@here/harp-texture-atlas-generator/src/cli-sprites-generator.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as program from "commander";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import { generateSprites, ProcessingOptions } from "./AtlasGenerator";
+import { getLogger, Logger, LogLevel } from "./Logger";
+
+const version = JSON.parse(
+    fs.readFileSync(path.join(__dirname, "..", "package.json"), "utf8")).version;
+const cpus: number = os.cpus() ? os.cpus().length : 4;
+
+const _defaultOutputDir: string = "./output";
+const _defaultWidth: number = 0;
+const _defaultHeight: number = 0; // height may be contrained to width with original aspect ratio
+const _defaultJobsNum: number = cpus;
+
+program
+    .version(version)
+    .description("CLI tool for generating texture atlases from multiple image sources.")
+    .usage("[options] URL")
+    .option("-i, --in [path]", "Input path")
+    .option("-o, --out [file]", "Output directory", _defaultOutputDir)
+    .option("-w, --width [num]", "Width of output sprite, set to zero leaves size unchanged " +
+        "or constraints to height (if set) while preserving aspect ratio", _defaultWidth)
+    .option("-h, --height [num]", "Height of output sprite, zero leaves original size " +
+        "or constraints to width (if set) while preserving aspect ratio", _defaultHeight)
+    .option("-v, --verbose", "Verbose mode")
+    .option("-j, --jobs [num]", "Number of processing threads", _defaultJobsNum)
+    .option("-c, --processConfig [path]", "Processing steps configuration json file")
+    .parse(process.argv);
+
+// Parse processing options
+const cliOptions = program.opts();
+const options: ProcessingOptions = {
+    input: cliOptions.in,
+    output: cliOptions.out,
+    width: parseInt(cliOptions.width, 10),
+    height: parseInt(cliOptions.height, 10),
+    verbose: cliOptions.verbose !== undefined ? true : false,
+    jobs: parseInt(cliOptions.jobs, 10),
+    processConfig: cliOptions.processConfig,
+};
+
+const logger: Logger = getLogger(options.verbose);
+
+if (options.verbose) {
+    logger.log(LogLevel.DEBUG, "\nVerbose mode active, params list:");
```

Comment:
Pro-tip. We're in JS world.
Don't boiler plate like this, just `logger.log('Options: ', options)`.

Smaller code, better code.

Reply:
I agree, just keep in mind that in the newest patch-set update there is much more formatting stuff, so when you run the cli it is clearly visible what, where and how the script is working, I would insist here to leave fine-tuned formatting just for better user experience and seeing what is happening there. Plain json-like output isn't so clear in my opinion, because options names are obviously some shorten version of parameters that may be not so straightforward.